app/models/concerns/foreman/thread_session.rb

```diff
@@ -27,11 +27,12 @@ module Cleaner
       def clear_thread
         if Thread.current[:user] && !Rails.env.test?
           Rails.logger.warn("Current user is set, but not expected. Clearing")
-          Thread.current[:user] = nil
+          User.current = nil
```

Comment:
👍 for this change: this way have a logic at once place handling setting the user current